### ISSUE TITLE
Variable in Platformer project hanging for a bit before beginning to update.

### ISSUE THREAD

**Issue by TommyOliverSays** [chrismaltby/gb-studio#1022]
Tying to script an action that lets you perform an action while in the air. Using the Player Fields plug-in but reporting the bug here on recommendation of the Discord server, incase th issue is with GB Studio itself.

Currently have a test project set up to fire a bullet when the player is in the air, as well as report the status of the "onGound" Bool when start is pressed. As you can see in the attached gif, for the first minute or so, the variable doesn't update, constantly reporting 0 when start is pressed regardless of player status, and fires bullets constantly. 

After ~ a minute, it suddenly starts working as expected, the variable updates when checked, and bullets only fire in the air.

On 3.02, on macOS Monterey. 

![gbstudio_test](https://user-images.githubusercontent.com/56774601/150706373-776ea303-8414-468c-a09c-5af2e2737e2a.gif)

**Comment by pau-tomas** [comment]
From the Discord conversation you're using an On Update script to store the is on ground variable 
![image](https://user-images.githubusercontent.com/54246642/150749713-83ac694b-fd66-4b4a-8345-1f4bd6362f36.png)

Is the actor that contains this on update script pinned?  otherwise it'll be out of the screen the moment you scroll to the right. Out of the screen actors are disabled and don't run their on update script.

**Comment by TommyOliverSays** [comment]
> From the Discord conversation you're using an On Update script to store the is on ground variable
> 
> <img alt="" width="300" src="https://user-images.githubusercontent.com/54246642/150749713-83ac694b-fd66-4b4a-8345-1f4bd6362f36.png">
> 
> Is the actor that contains this on update script pinned? otherwise it'll be out of the screen the moment you scroll to the right. Out of the screen actors are disabled and don't run their on update script.

I’m using a pinned hidden actor as a Game Manager, yes. Would it help to link the project file?

**Comment by pau-tomas** [comment]
Yes, please, include a zip of the project.

**Comment by TommyOliverSays** [comment]
Pinned sprite in the corner is in the top left corner with the variable update. Scene has the Attach to button script.

[test.zip](https://github.com/chrismaltby/gb-studio/files/7924701/test.zip)

**Comment by TommyOliverSays** [comment]
Another observation: It seems the variable starts working not after a set amount of time, but after moving off and then back on the first platform. I tired jumping back on the first platform from each side, and as soon as the Player lands on that one in particular, the variable starts working as expected.

**Comment by TommyOliverSays** [comment]
> Another observation: It seems the variable starts working not after a set amount of time, but after moving off and then back on the first platform. I tired jumping back on the first platform from each side, and as soon as the Player lands on that one in particular, the variable starts working as expected.

Okay, it has to do with the hidden Game Manager sprite that's updating the variable. If you disable the On Init Hide Actor action, everything works as expected immediately. 

Maybe the reason jumping on that first platform was triggering it working, was that the hidden game manager actor's location was back on screen. It's pinned so it should be moving, but jumping back on that first platform would have where it's initial location in the Scene is (I'm not a programmer so I'm just spitballing haha)

**Comment by pau-tomas** [comment]
Indeed. Hidden actors aren't suppose to be running the on update script, except when #884 happens.

**Comment by TommyOliverSays** [comment]
> Indeed. Hidden actors aren't suppose to be running the on update script, except when #884 happens.

Huh... okay. It looks like I've run into the same issue as #884 then, except in my case, it made it work how I intended, because I was using the On Update function incorrectly.

The only odd thing is the hidden sprite was pinned to the screen, so it should never have left and come back to the viewport. I'd assume pinning an actor is essentially in the On Update function under the hood?
